app/v1/controllers/requestParsers/SubmitReturnRequestParser.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers
+
+import javax.inject.Inject
+import uk.gov.hmrc.domain.Vrn
+import v1.controllers.requestParsers.validators.SubmitValidator
+import v1.models.request.submit.{SubmitRawData, SubmitRequest, SubmitRequestBody}
+
+class SubmitReturnRequestParser @Inject()(val validator: SubmitValidator)
+  extends RequestParser[SubmitRawData, SubmitRequest] {
+
+  override protected def requestFor(data: SubmitRawData): SubmitRequest = {
+    SubmitRequest(Vrn(data.vrn), data.body.json.as[SubmitRequestBody])
```

Comment:
```suggestion
    SubmitRequest(Vrn(data.vrn), data.body.json)
```

Since we send the data down to DES as json anyway, and we can't reach this point without the json being valid we can just use the request json here. This skips an unnecessary conversion from json -> model -> back to json, and also means you can just remove the writes from the request body model.

Reply:
worth mentioning that this is just personal preference. there's nothing wrong with your implementation

Reply:
In this case, we have to convert as model and then use writes to convert to JSON as Des request JSON is different from MTD request JSON

Reply:
Ah, I hadn't noticed that the field names were slightly different between the two request models.